Automattic/woocommerce-payments
# Share selected currency state across account overview page components

Resolves #5753
Resolves #5754 

#### Changes proposed in this Pull Request

- The selected currency state is now stored as a URL query param `selected_currency`. This allows the correct currency to remain selected when using the browser "back" button.
- This is updated when the Account Balances currency tab is selected.
- Two new hooks have been added to `clients/overview/hooks.ts`
  - `useSelectedCurrency`: for retrieving and updating the selected currency.
  - `useSelectedCurrencyOverview`: to get the selected currency overview.
- These hooks have been integrated into the Account Balances and Deposit Overview cards.


Screen recording of the state syncing between the cards and after a page refresh:


https://user-images.githubusercontent.com/3147296/231020380-ca9e157e-4ef0-4f62-bd3e-6d3c421e0c39.mov


#### Testing instructions





1. Enable the `_wcpay_feature_simplify_deposits_ui` feature flag.
To enable, run `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
5. Add more than one bank account currency to your store. [See public docs](https://woocommerce.com/document/woocommerce-payments/deposits/deposit-currencies/#adding-multiple-bank-accounts).
6. Enable multi-currencies for more than one of your bank accounts.
7. Purchase some products in each currency, some using the "bypass pending" card number to add funds straight to "available funds": `4000000000000077`.
8. Select each currency in the Account Overview card and ensure the correct currency and amounts are rendered.
9. Viewing the deposit history table will require previous deposits for the selected currency. To achieve this, purchase items with the "available funds" card above, set the deposit schedule to "daily" and wait 1+ days.
10. Refreshing the page should persist the selected currency, once loading has completed.

-------------------


**Post merge**
